Common/cpp/Tools/JSISerializer.cpp

```diff
@@ -5,23 +5,53 @@
 #endif
 
 #include <cxxabi.h>
+#include <iostream>
 #include <sstream>
 
-static inline bool
-isInstanceOf(jsi::Runtime &rt, jsi::Object &object, const std::string &type) {
-  return object.instanceOf(
-      rt, rt.global().getPropertyAsFunction(rt, type.c_str()));
+static inline std::string getObjectTypeName(
+    jsi::Runtime &rt,
+    const jsi::Object &object) {
+  return object.getPropertyAsObject(rt, "constructor")
+      .getProperty(rt, "name")
+      .toString(rt)
+      .utf8(rt);
+}
+
+static inline bool isInstanceOf(
+    jsi::Runtime &rt,
+    const jsi::Object &object,
+    const std::string &type) {
+  return getObjectTypeName(rt, object) == type;
+}
+
+static inline bool isOneOfSupportedTypes(
+    jsi::Runtime &rt,
+    const jsi::Object &object,
+    std::vector<std::string> supportedTypes) {
```

Comment:
Avoid copying of vector

```suggestion
    const std::vector<std::string>& supportedTypes) {
```